x/signal/keeper.go

```diff
@@ -102,6 +126,9 @@ func (k Keeper) VersionTally(ctx context.Context, req *types.QueryVersionTallyRe
 	iterator := store.Iterator(nil, nil)
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
+		if bytes.Equal(iterator.Key(), types.UpgradeKey) {
+			continue
+		}
```

Comment:
You don't need to include the if statement if you want to start the iterator after the upgrade key (since the upgrade key is presumably the first key in the kv store).

Reply:
how do I start the iterator at the first key _after_ the upgrade key? I expect this will start the iterator _at_ the upgrade key:

```
	iterator := store.Iterator(types.UpgradeKey, nil)
```

and I don't have a constant for the first value _after_ the upgrade key. If we start an iterator like this, we'll need to skip the first value which seems just as janky as the explicit conditional:

```go
		if bytes.Equal(iterator.Key(), types.UpgradeKey) {
			continue
		}
```

Reply:
What's wrong with adding a constant like:

```go
var FirstSignalKey = []byte{0x000} 
```

and start the iterator there

Reply:
hmm nothing seems wrong with that but I wasn't sure which would be logically after `[]byte{0x00}`: A: `[]byte{0x000}` or B: `[]byte{0x01}`.

 I wonder if we should reserve more keys at the beginning of the range in case we want the signal keeper to persist other things in the state store. Although if we do that, we would need a state migration at which point we can reserve more keys so may not need them now.

Reply:
3e791ddd25518cfa34cc5068f0f06026e478babd